sql/core/src/main/scala/org/apache/spark/sql/execution/PartitionedFileUtil.scala

```diff
@@ -64,13 +64,14 @@ object PartitionedFileUtil {
       offset: Long,
       length: Long): Array[String] = {
     val candidates = blockLocations.map {
-      // The fragment starts from a position within this block
+      // The fragment starts from a position within this block. It handles the case where the
+      // fragment is fully contained in the block.
       case b if b.getOffset <= offset && offset < b.getOffset + b.getLength =>
         b.getHosts -> (b.getOffset + b.getLength - offset).min(length)
 
       // The fragment ends at a position within this block
-      case b if offset <= b.getOffset && offset + length < b.getLength =>
-        b.getHosts -> (offset + length - b.getOffset).min(length)
+      case b if b.getOffset <= offset + length && offset + length < b.getOffset + b.getLength =>
```

Comment:
Shall we remove the equality?
```scala
- case b if b.getOffset <= offset + length && offset + length < b.getOffset + b.getLength =>
+ case b if b.getOffset < offset + length && offset + length < b.getOffset + b.getLength =>
```

In case of `b.getOffset = offset + length`, `offset + length - b.getOffset` is `0L` effectively. So, line 74 becomes `b.getHosts -> 0L` effectively which is identical with line 82.

Reply:
It won't matter either way, but yeah if `b.getOffset == offset + length`, you could say the fragment doesn't actually end in the block; it ends just before it. Just for symmetry it might be OK to keep it this way, but I don't feel strongly about it.

Reply:
Equality removed.